src/state/pools/hooks.ts

```diff
@@ -23,22 +23,28 @@ import {
 import { State, DeserializedPool, VaultKey } from '../types'
 import { transformPool } from './helpers'
 import { fetchFarmsPublicDataAsync, nonArchivedFarms } from '../farms'
+import { useCurrentBlock } from '../block/hooks'
 
 export const useFetchPublicPoolsData = () => {
   const dispatch = useAppDispatch()
 
-  useSlowRefreshEffect(() => {
-    const fetchPoolsDataWithFarms = async () => {
-      const activeFarms = nonArchivedFarms.filter((farm) => farm.pid !== 0)
-      await dispatch(fetchFarmsPublicDataAsync(activeFarms.map((farm) => farm.pid)))
-      batch(() => {
-        dispatch(fetchPoolsPublicDataAsync())
-        dispatch(fetchPoolsStakingLimitsAsync())
-      })
-    }
+  useSlowRefreshEffect(
+    (currentBlock) => {
+      const fetchPoolsDataWithFarms = async () => {
+        const activeFarms = nonArchivedFarms.filter((farm) => farm.pid !== 0)
+        await dispatch(fetchFarmsPublicDataAsync(activeFarms.map((farm) => farm.pid)))
+        batch(() => {
+          dispatch(fetchPoolsPublicDataAsync(currentBlock))
+          dispatch(fetchPoolsStakingLimitsAsync())
+        })
+      }
 
-    fetchPoolsDataWithFarms()
-  }, [dispatch])
+      if (currentBlock > 0) {
```

Comment:
If it is depends on currentBlock, which will be `0` at first and wait for slow interval. We should handle it